### PR TITLE
Knative Generator Rewrite

### PR DESCRIPTION
**Pull request by Fabian-K**
[knative]
- update generator to use shared logic
- replace knative-annotator with shared model-annotator
- remove the generated *Schema class

[tekton]
- update to knative package change

[common]
- add editorconfig for *.go (although this is not applied to all go files yet)

As always, I´m looking forward to feedback!

Thanks,
Fabian

### CODE REVIEW

**Comment by rohanKanojia** [comment]
ok to test

**Comment by Fabian-K** [comment]
Hey, thanks for already looking into this! I´d suggest pausing this for now, I´d rebase it on top of https://github.com/fabric8io/kubernetes-client/pull/2160

File: extensions/knative/generator/cmd/generate/generate.go [hunk 1]
```diff
@@ -16,169 +16,91 @@
...
-	}
 
-	typeMap := map[reflect.Type]reflect.Type{
-		reflect.TypeOf(time.Time{}): reflect.TypeOf(""),
-		reflect.TypeOf(struct{}{}):  reflect.TypeOf(""),
+	// the CRD List types for which the model should be generated
+	// no other types need to be defined as they are auto discovered
+	crdLists := []reflect.Type{
+
+		// serving v1
```
Comment (rohanKanojia) [inline extensions/knative/generator/cmd/generate/generate.go:37]: Umm, why have you removed all resources from `duck v1beta1`. Are they not part of Knative model?

Comment (Fabian-K) [reply extensions/knative/generator/cmd/generate/generate.go:36]: 2 things:
- all types that are only used internally are auto-discovered by the generator. Only CRDs need to be passed in as a starting point.
- AFAIK everything related to duck is not a CRD exposed by knative. Instead that is somehow used internally for duck typing. I might be wrong on this, do you have other information?

Comment (rohanKanojia) [reply extensions/knative/generator/cmd/generate/generate.go:36]: Umm, I don't have much idea about Knative. I think it's okay if duck classes were not exposed earlier.

**Comment by rohanKanojia** [comment]
[merge]

**Pull request merged**